# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/android/Platform.cpp
# Commit: 3d0747ce0174
# Full Hash: 3d0747ce0174e52cf7196dd4638c6307f8d88d0d
# Author: James Teh <jteh@mozilla.com>
# Date: 2023-11-30 04:12:04
# Regressor Bug: 1862802
# File Overlap Count: 1
# Description:
#   Bug 1862802 part 2: Support find in page on Android using only caret events instead of converting them to virtual cursor change events. r=eeejay
#   
#   Depends on D192642
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D192643
# ==============================================================================

diff -r d1558d374cef -r 3d0747ce0174 accessible/android/Platform.cpp
--- a/accessible/android/Platform.cpp	Tue Nov 28 22:03:19 2023 +0000
+++ b/accessible/android/Platform.cpp	Tue Nov 28 22:03:19 2023 +0000
@@ -146,10 +146,23 @@
                                   bool aFromUser) {
   RefPtr<SessionAccessibility> sessionAcc =
       SessionAccessibility::GetInstanceFor(aTarget);
+  if (!sessionAcc) {
+    return;
+  }
 
-  if (sessionAcc) {
-    sessionAcc->SendTextSelectionChangedEvent(aTarget, aOffset);
+  if (!aTarget->IsDoc() && !aFromUser && !aIsSelectionCollapsed) {
+    // Pivot to the caret's position if it has an expanded selection.
+    // This is used mostly for find in page.
+    Accessible* leaf = TextLeafPoint::GetCaret(aTarget).ActualizeCaret().mAcc;
+    MOZ_ASSERT(leaf);
+    if (Accessible* result = AccessibleWrap::DoPivot(
+            leaf, java::SessionAccessibility::HTML_GRANULARITY_DEFAULT, true,
+            true)) {
+      sessionAcc->SendAccessibilityFocusedEvent(result);
+    }
   }
+
+  sessionAcc->SendTextSelectionChangedEvent(aTarget, aOffset);
 }
 
 void a11y::PlatformTextChangeEvent(Accessible* aTarget, const nsAString& aStr,
